en/lessons/basics/collections.md

```diff
@@ -3,120 +3,224 @@ version: 1.3.1
+
+```haskell
+instance Semigroup [a] where
+  (<>) = (++)
+```
+
+The keywords aren't important, but it should give you an intuition for when you see `<>` which just means concatenation for a specific type!
 
-### List Subtraction
+### List Difference
 
-Support for subtraction is provided via the `--/2` operator; it's safe to subtract a missing value:
+Support for difference is provided via the `\\` operator; it's safe to subtract a missing value:
 
-```elixir
-iex> ["foo", :bar, 42] -- [42, "bar"]
-["foo", :bar]
+```haskell
+ghci> import Data.List
+ghci> ["foo", "bar", "baz"] \\ ["quux", "bar"]
+["foo","baz"]
 ```
 
-Be mindful of duplicate values.
-For every element on the right, the first occurrence of it gets removed from the left:
+__Note__: `import` syntax may be new to you, but this allows us to pull in libraries into the repl, and fortunately for us GHC ships with a list module in its base library!
 
-```elixir
-iex> [1,2,2,3,2,3] -- [1,2,3,2]
-[2, 3]
+Be mindful of duplicate values. For every element on the right, the first occurrence of it gets removed from the left:
+
+```haskell
+ghci> ["foo", "bar", "baz", "bar"] \\ ["quux", "bar"]
+["foo","baz","bar"]
 ```
 
-**Note:** List subtraction uses [strict comparison](../basics/#comparison) to match the values. For example:
+List difference uses an [equality](../basics/#comparison) instance for your specific type, to match values. We can see this in the type.
 
-```elixir
-iex> [2] -- [2.0]
-[2]
-iex> [2.0] -- [2.0]
-[]
+```haskell
+ghci> :t (\\)
+(\\) :: Eq a => [a] -> [a] -> [a]
 ```
 
+The `Eq a` on the left hand side of the constraint arrow `=>` means that all `a`'s on the right must provide an implementation of equality. This allows us to compare the elements in the list and subtract them.
+
 ### Head / Tail
 
 When using lists, it is common to work with a list's head and tail.
 The head is the list's first element, while the tail is a list containing the remaining elements.
-Elixir provides two helpful functions, `hd` and `tl`, for working with these parts:
 
-```elixir
-iex> hd [3.14, :pie, "Apple"]
-3.14
-iex> tl [3.14, :pie, "Apple"]
-[:pie, "Apple"]
+Haskell provides two helpful functions, `head` and `tail`, for working with these parts:
+
+```haskell
+ghci> head ["Orange", "Banana", "Apple"]
+"Orange"
+ghci> tail ["Orange", "Banana", "Apple"]
+["Banana","Apple"]
 ```
 
-In addition to the aforementioned functions, you can use [pattern matching](../pattern-matching/) and the cons operator `|` to split a list into head and tail. We'll learn more about this pattern in later lessons:
+Unfortunately these functions reveal an ugly part of the language's base library, some of the functions are partial. This means that they do not cover the full domain of possible inputs.
```

Comment:
It would be nice to give an example here.

Reply:
whoops, the example below was supposed to be just this!